### PR TITLE
added automatic cypress login

### PR DESCRIPTION
**Pull request by naphets123**

### CODE REVIEW

File: cypress/integration/1_1_LoginTests.js [hunk 1]
```diff
@@ -5,21 +5,31 @@ context('Login tests', () => {
...
+    cy.get("input[id='1-email']").type(`${userMail}`);
+    cy.get("input[name ='password']").type(`${userPassword}`);
+    cy.get("button[class='auth0-lock-submit']").click();
+  };
 
   it('Login test (Admin)', () => {
     console.log(config.testAdmin);
 
     console.log(config);
+    //Auth0_login(config.testAdmin,config.testPwd);
```
Comment (HaGuesto) [inline cypress/integration/1_1_LoginTests.js:28]: remove comment

File: cypress/integration/1_1_LoginTests.js [hunk 2]
```diff
@@ -5,21 +5,31 @@ context('Login tests', () => {
   let config = getLoginConfiguration();
 
   // Login commands trhough login page
+  /*
```
Comment (HaGuesto) [inline cypress/integration/1_1_LoginTests.js:8]: remove comment

File: cypress/support/session.js [hunk 3]
```diff
@@ -39,3 +59,4 @@ Cypress.Commands.add("loginAsCoordinator", () => {
     let config = getLoginConfiguration();
     loginWithAjax(config.testCoordinator, config.testPwd);
 });
+
```
Comment (HaGuesto) [inline cypress/support/session.js:62]: remove line

File: cypress/support/session.js [hunk 4]
```diff
@@ -1,21 +1,41 @@
...
+    cy.log(`Logging in as ${username}`);
+        const options = {
+            method: 'POST',
+            url: Cypress.env('auth_url'),
+            body: {
+                grant_type: 'password',
+                username: username,
+                password: password,
+                audience: Cypress.env('auth_audience'),
+                scope: 'openid profile email',
```
Comment (HaGuesto) [inline cypress/support/session.js:15]: is that supposed to be that general?

File: cypress/support/session.js [hunk 5]
```diff
@@ -1,21 +1,41 @@
...
+                client_id: Cypress.env('auth_client_id'),
+                client_secret: Cypress.env('auth_client_secret'),
+            },
+        };
+        cy.request(options).then(({ body }) => {
+            const { access_token, expires_in, id_token } = body;
+            console.log(access_token);
```
Comment (HaGuesto) [inline cypress/support/session.js:22]: please remove

File: cypress/support/session.js [hunk 6]
```diff
@@ -1,21 +1,41 @@
...
+            cy.request({
+                method: "POST",
+                url: '/ajax.php?file=cypresslogin',
+                body: {
+                    access_token: access_token,
+                    id_token: id_token
+
```
Comment (HaGuesto) [inline cypress/support/session.js:29]: remove line

File: cypress/support/session.js [hunk 7]
```diff
@@ -1,21 +1,41 @@
...
+                },
+                form: true
```
Comment (HaGuesto) [inline cypress/support/session.js:31]: what is this for?

File: cypress/support/session.js [hunk 8]
```diff
@@ -1,21 +1,41 @@
...
+            }).then(response => {
+                expect(response.status).to.eq(200);
+                expect(response.body.message).to.be.empty;
```
Comment (HaGuesto) [inline cypress/support/session.js:34]: can you check  the  `success` parameter instead of the `message` parameter?

File: library/lib/auth0.php [hunk 9]
```diff
@@ -1,8 +1,11 @@
 <?php
 
 require 'vendor/autoload.php';
+
```
Comment (HaGuesto) [inline library/lib/auth0.php:4]: remove line

File: library/lib/auth0.php [hunk 10]
```diff
@@ -1,8 +1,11 @@
...
 use Auth0\SDK\Auth0;
 
+//use Auth0\SDK\API\Authentication;
```
Comment (HaGuesto) [inline library/lib/auth0.php:7]: remove lines

File: library/lib/auth0.php [hunk 11]
```diff
@@ -12,6 +15,26 @@
...
+if (isset($_POST['access_token'])) {
+    $auth0->setAccessToken($_POST['access_token']);
+    if (isset($_POST['refresh_token'])) {
+        $auth0->setRefreshToken($_POST['refresh_token']);
+    }
+    if (isset($_POST['id_token'])) {
+        $auth0->setIdToken($_POST['id_token']);
+    }
+    $curlcon = curl_init();
+    curl_setopt($curlcon, CURLOPT_URL, 'https://dev-tgx41z7g.eu.auth0.com/userinfo');
```
Comment (HaGuesto) [inline library/lib/auth0.php:27]: can u replace the url with the `$settings`

File: library/lib/auth0.php [hunk 12]
```diff
@@ -12,6 +15,26 @@
 
 $userInfo = $auth0->getUser();
```
Comment (HaGuesto) [inline library/lib/auth0.php:17]: could u add a command that this is only for cypress tests.
Could you also add a check that this only works for certain urls requesting like `staging.boxwise.co` and `localhost:8...`

File: cypress/support/session.js (hunk 4 above)
Comment (naphets123) [reply cypress/support/session.js:15]: The way we currently use it we will use the email as a username and pull the profile data on the php side for authorization. So at the moment i guess so

File: cypress/support/session.js (hunk 7 above)
Comment (HaGuesto) [reply cypress/support/session.js:30]: @naphets123

File: cypress/support/session.js (hunk 8 above)
Comment (HaGuesto) [reply cypress/support/session.js:33]: @naphets123

File: cypress/support/session.js (hunk 7 above)
Comment (HaGuesto) [reply cypress/support/session.js:30]: remove if not needed https://docs.cypress.io/api/commands/request.html#Arguments

**Comment by HaGuesto** [comment]
[Integrate PHP app with auth0 for identity management (authentication) only #php #extra-large](https://trello.com/c/8KEgl3nv/503-integrate-php-app-with-auth0-for-identity-management-authentication-only-php-extra-large)

**Comment by aerinsol** [comment]
@HaGuesto what's the story with this PR?

**Pull request closed without merging**